src/app/getting-started/getting-started.component.ts

```diff
@@ -0,0 +1,187 @@
+
+  authGitHub: boolean = false;
+  authOpenShift: boolean = false;
+  gettingStartedCompleted: boolean = false;
+  gitHubLinked: boolean = false;
+  loggedInUser: User;
+  openShiftLinked: boolean = false;
+  registrationCompleted: boolean = false;
+  showGettingStarted: boolean = false;
+  username: string;
+  usernameInvalid: boolean = false;
+
+  constructor(
+      private auth: AuthenticationService,
+      private gettingStartedService: GettingStartedService,
+      private logger: Logger,
+      private providerService: ProviderService,
+      private notifications: Notifications,
+      private router: Router,
+      private userService: UserService) {
+    userService.loggedInUser.subscribe(user => {
+      if (user === undefined || user.attributes === undefined) {
+        return;
+      }
+      this.loggedInUser = user;
+      this.username = this.loggedInUser.attributes.username;
+      this.registrationCompleted = (user as ExtUser).attributes.registrationCompleted;
+    });
+    auth.gitHubToken.subscribe(token => {
+      this.gitHubLinked = (token !== undefined && token.length !== 0);
+    });
+    auth.openShiftToken.subscribe(token => {
+      this.openShiftLinked = (token !== undefined && token.length !== 0);
+    });
+    this.isGettingStarted().subscribe(val => {
+      this.showGettingStarted = val;
+      if (!this.showGettingStarted) {
+        this.routeToHome();
+      }
+    });
+  }
+
+  ngOnInit() {
+  }
+
+  /**
+   * Helper to test if connect accounts button should be disabled
+   *
+   * @returns {boolean}
+   */
+  get isConnectAccountsDisabled(): boolean {
+    return !(this.authGitHub && !this.gitHubLinked || this.authOpenShift && !this.openShiftLinked)
+      || (this.gitHubLinked && this.openShiftLinked);
+  }
+
+  /**
+   * Helper to test if the successful state panel should be shown
+   *
+   * @returns {boolean} If the user has completed the getting started page
+   */
+  get isSuccess(): boolean {
+    return !this.usernameInvalid && this.gitHubLinked && this.openShiftLinked;
+  }
+
+  // Actions
+
+  /**
+   * Link GitHub and/or OpenShift accounts
+   */
+  connectAccounts(): void {
+    if (this.authGitHub && !this.gitHubLinked && this.authOpenShift && !this.openShiftLinked) {
+      this.providerService.linkAll(window.location.origin + "/_gettingstarted?wait=true");
+    } else if (this.authGitHub && !this.gitHubLinked) {
+      this.providerService.linkGitHub(window.location.origin + "/_gettingstarted?wait=true");
+    } else if (this.authOpenShift && !this.openShiftLinked) {
+      this.providerService.linkOpenShift(window.location.origin + "/_gettingstarted?wait=true");
+    }
+  }
+
+  /**
+   * Helpfer to route to home page
+   */
+  routeToHome(): void {
+    this.router.navigate(['/', '_home']);
+  }
+
+  /**
+   * Update username
+   *
+   * Note: This can only be done once per the user patch API
+   */
+  updateUsername(): void {
+    this.usernameInvalid = !this.isUsernameValid();
```

Comment:
I'm just ensuring what I passed to the API is returned to the users patch API. I don't have anyway to test uniqueness client-side. Unless there is an API?